### ISSUE TITLE
SQLite database locked

### ISSUE THREAD

**Issue by SlimeDog** [Rsl1122/Plan-PlayerAnalytics#319]
Spigot 1.12.2
PLAN 4.0.1

I'm getting a lot of these in the log:
```
[16:04:16 ERROR]: [Plan] main.java.com.djrapitops.plan.systems.processing.CommandProcessor Caught org.sqlite.SQLiteException: [SQLITE_BUSY]  The database file is locked (database is locked). It has been logged to Errors.txt
[16:04:16 ERROR]: [Plan] main.java.com.djrapitops.plan.systems.processing.CommandProcessor Caught org.sqlite.SQLiteException: [SQLITE_BUSY]  The database file is locked (database is locked). It has been logged to Errors.txt
```

Errors.txt says:
```
main.java.com.djrapitops.plan.systems.processing.CommandProcessor Caught org.sqlite.SQLiteException: [SQLITE_BUSY]  The database file is locked (database is locked)
  org.sqlite.core.DB.newSQLException(DB.java:909)
  org.sqlite.core.DB.newSQLException(DB.java:921)
  org.sqlite.core.DB.execute(DB.java:822)
  org.sqlite.core.DB.executeUpdate(DB.java:863)
  org.sqlite.jdbc3.JDBC3PreparedStatement.executeUpdate(JDBC3PreparedStatement.java:99)
  org.apache.commons.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:98)
  org.apache.commons.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:98)
  main.java.com.djrapitops.plan.database.tables.CommandUseTable.commandUsed(CommandUseTable.java:119)
  main.java.com.djrapitops.plan.systems.processing.CommandProcessor.process(CommandProcessor.java:26)
  main.java.com.djrapitops.plan.systems.queue.ProcessConsumer.consume(ProcessingQueue.java:55)
  main.java.com.djrapitops.plan.systems.queue.ProcessConsumer.consume(ProcessingQueue.java:40)
  main.java.com.djrapitops.plan.systems.queue.Consumer.run(Consumer.java:35)
  com.djrapitops.plugin.task.RunnableFactory$1.run(RunnableFactory.java:37)
  org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:71)
  org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:52)
  java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  java.lang.Thread.run(Thread.java:748)
```

I'm doing a lot of manual reload and analyze as I fiddle with the content of config.yml and server.html. That may or may not be the cause, but hopefully SQLite file locks can be managed.

**Comment by Rsl1122** [comment]
Constant issue, working on it.

**Comment by Rsl1122** [comment]
Possible cause: Way Ahead Logging not enabled on old sqlite db files.

How to check:
Are the following files next to the database.db:
- database.db-shm
- database.db-wal

Possible fixes:
- run /plan m backup sqlite
- rename the 3 new files to match database (.db .db-shm .db-wal)

**Comment by SlimeDog** [comment]
Sample of one session after database restore. All OK. Bonus: no extra plan_sessions or plan_world_times records. I'll keep testing.

**Comment by Rsl1122** [comment]
> New database disk image is malformed.
This occurs if the .db-wal and .db-shm do not match the .db file, so I guess forcibly renaming them made them unmatch the database.

I'm hoping the fix for #309 fixes this issue

**Comment by SlimeDog** [comment]
I haven't seen this issue in 4.0.2.
